json/writer.go

```diff
@@ -1,32 +1,16 @@
 // SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
 
-package spdx_json
+package json
 
 import (
 	"encoding/json"
-	"github.com/spdx/tools-golang/spdx/v2_3"
 	"io"
 
-	"github.com/spdx/tools-golang/spdx/v2_2"
+	"github.com/spdx/tools-golang/spdx/common"
 )
 
-// Save2_2 takes an SPDX Document (version 2.2) and an io.Writer, and writes the document to the writer in JSON format.
-func Save2_2(doc *v2_2.Document, w io.Writer) error {
-	buf, err := json.Marshal(doc)
-	if err != nil {
-		return err
-	}
-
-	_, err = w.Write(buf)
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-// Save2_3 takes an SPDX Document (version 2.2) and an io.Writer, and writes the document to the writer in JSON format.
-func Save2_3(doc *v2_3.Document, w io.Writer) error {
+// Write takes an SPDX Document and an io.Writer, and writes the document to the writer in JSON format.
+func Write(doc common.AnyDocument, w io.Writer) error {
```

Comment:
probably a next step is for all output encoders to provide an option of the version requested, we'll open a separate issue for this, not a blocker.

Reply:
the `Write` functions accept all versions and output the corresponding file, e.g. a v2_2.Document will generate a 2.2 version if passed in. I'd like this to be more strongly typed. The only way I could think of doing this is to create an interface with an arbitrary function, and implement it for each `Document` types...